src/PiWeb.Import.Sdk/ImportFiles/IImportGroup.cs

```diff
@@ -28,37 +27,37 @@ public interface IImportGroup
     /// <summary>
     /// The primary file with which the grouping was started. This file has to be recognized by the format.
     /// </summary>
-    IFileSource PrimaryFile { get; }
+    IImportFile PrimaryFile { get; }
 
     /// <summary>
     /// The list of additional files recognized by the import format and added using <see cref="AddFile"/>.
     /// The list does not include <see cref="PrimaryFile"/>. These files will not be offered to other import formats.
     /// </summary>
-    ICollection<IFileSource> AdditionalFiles { get; }
+    IEnumerable<IImportFile> AdditionalFiles { get; }
 
     /// <summary>
     /// The list of files recognized by the import format including <see cref="PrimaryFile"/>.
     /// These files will not be offered to other import formats.
     /// </summary>
-    ICollection<IFileSource> AllFiles { get; }
+    IEnumerable<IImportFile> AllFiles { get; }
 
     /// <summary>
-    /// Adds the <paramref name="fileSource"/> to this <see cref="IImportGroup"/>. 
+    /// Adds the <paramref name="importFile"/> to this <see cref="IImportGroup"/>. 
     /// These files will not be offered to other import formats. 

```

Comment:
**This file** will not be offered to other import formats.

Reply:
I have rewritten this completely. I don't think this is a good place to explain the filtering chain so I left it out.